Creates a page summarizing all Elastic's AI-powered features

benironside opened the pull request:
Fixes elastic/docs-content-internal/issues/455 by creating a new page that lists all our AI-powered features. 

This PR also:
- Creates an AI-powered features section in the explore-analyze book (new home for the new doc and IMO improved IA for the AI pages in the book).
- Makes some related small edits to the pages describing those features in detail.

Also, reviewers, this is a minor point but what do you think of adding "AI" to our [glossary](https://www.elastic.co/docs/reference/glossary) and linking to this PR's new page?

florent-leborgne reviewed (commented):
This is a super nice start. I'd like this page (or set of pages) to do the following:
- List the AI features themselves (AI Assistants, Agent Builder) and clarify the link between these and AI connectors (and with EIS for the Elastic Managed LLM)
- List other features that rely on these AI features, and specify if this is optional, by default, etc -> This is important for users to make conscious choices about their config and about permissions of their own users, especially as they'll likely want to control pricing/token usage under control.
- Links to pricing pages for our default Elastic Managed LLM (at least) or to a page that focuses on AI features-related pricing impact

So that this inventory also becomes a source of understanding of how all of these relate to each other. We can chat about this at our sync :)

mdbirnstiehl commented on explore-analyze/ai-features/ai-features.md:
````diff
@@ -0,0 +1,155 @@
+stack: preview 9.1, ga 9.2
+```
+You can [generate Grok patterns](/solutions/observability/streams/management/extract/grok.md#streams-grok-patterns) using AI instead of writing them by by hand. 
+
````
I thought of one other AI component to streams, you can also partition wired streams based on AI suggestions. More info here. https://www.elastic.co/docs/solutions/observability/streams/management/partitioning. If you want some assistance with writing up the blurbs about these, I can help tomorrow.

benironside replied:
Feel free to make whatever edits you see fit

szabosteve reviewed (approved):
I left two small nits and a suggestion to drop some technical details that I think are too low-level for this overview. I really like the clarity and the structure of this page! It's great to see these features listed in one place. Everything that is on the page right now is LGTM. I think it would still be nice to add some more pieces to the page. Basically, all that Liam mentioned above. Some suggestions regarding the structure for the new content:

* I would suggest adding two or three new sections:
  * Elastic inference or something similar (before AI-powered search), which could link to [this page](https://www.elastic.co/docs/explore-analyze/elastic-inference) with a description like this: "Inference is a process of using a machine learning trained model to make predictions or operations - such as text embedding, or reranking - on your data." This subsection could mention EIS and the Inference API as the two main ways to use Elastic Inference.
  * NLP models: this section could contain the [built-in NLP models](https://www.elastic.co/docs/explore-analyze/machine-learning/nlp/ml-nlp-built-in-models) with ELSER highlighted, and the [trained models deployed in your cluster](https://www.elastic.co/docs/explore-analyze/machine-learning/nlp/ml-nlp-overview). 
  * I think Elastic Managed LLM could have its own section. If you think it's too much, I would add it to the Elastic Inference section.

@leemthompo WDYT?

benironside commented:
I just pushed some updates to the security section. I mostly made the it more concise. Also changed the Attack Discovery docs to remove the confusing line you pointed out in your comment, Florent. I considered adding more info to each feature about how you need to configure an LLM connector to use it (this applies to all the features in the security section), but I decided not to, in favor of consistency across the solution sections (also since we point out at the top of the page that many of the features need LLM connectors it seemed a bit redundant). 

I also added a brief section at the top of the Security section that addresses some of the points identified by Florent.

florent-leborgne commented on explore-analyze/ai-features/ai-features.md:
````diff
@@ -0,0 +1,178 @@
+```
+Natural Language Processing (NLP) enables you to analyze natural language data and make predictions.
+
+Elastic offers a range of [built-in NLP models](/explore-analyze/machine-learning/nlp/ml-nlp-built-in-models.md) such as the Elastic-trained [ELSER](/explore-analyze/machine-learning/nlp/ml-nlp-elser.md). You can also [deploy custom models](/explore-analyze/machine-learning/nlp/ml-nlp-overview.md).
````
@szabosteve Should we more explicitly call out "Machine learning" in this section?

I'm thinking it may be a little clearer to present this section with something like this?

**Hi these are AI-powered capabilities or features augmented with AI available to you without requiring a specific solution context:**
- EIS. This is EIS... It powers:
  - the Elastic Managed LLM, which is xxx and that you can use in xxx
  - ELSER, which is is an NLP model trained by Elastic that enables you to perform [semantic search](https://www.elastic.co/docs/solutions/search/semantic-search) with Elastic machine learning.

benironside commented on explore-analyze/ai-features/ai-features.md:
```diff
@@ -0,0 +1,178 @@
+## Requirements
+
+- To use Elastic's AI-powered features, you need an appropriate license and feature tier. These vary by solution and feature. Refer to each feature's documentation to learn more.
+- Most features require at least one working LLM connector. To learn about setting up large language model (LLM) connectors used by AI-powered features, refer to [](/solutions/security/ai/set-up-connectors-for-large-language-models-llm.md).
```
We don't yet. But as soon as we merge this page, @mdbirnstiehl and I are going to finish https://github.com/elastic/docs-content-internal/issues/487 by moving the guides that are in the security solution into the section that this PR creates.

leemthompo commented:
I pushed a commit to clarify ES solution versus platform and some minor edits: https://github.com/elastic/docs-content/pull/3768/commits/ad81ecb1da7a41995222755f8b633a75e89db84a

The diff isn't great because GitHub is trying to be too smart 😄
